clients/venice-thin-client/src/main/java/com/linkedin/venice/client/store/RetriableStoreClient.java

```diff
@@ -60,7 +91,8 @@ public CompletableFuture<Map<K, V>> batchGet(Set<K> keys) throws VeniceClientExc
 
   private <T> CompletableFuture<T> executeWithRetry(Supplier<CompletableFuture<T>> supplier, RequestType requestType) {
     CompletableFuture<T> retryFuture = new CompletableFuture<>();
-    supplier.get().whenComplete((T val, Throwable throwable) -> {
+
+    supplier.get().whenCompleteAsync((T val, Throwable throwable) -> {
```

Comment:
The underlying d2 service is using 10s timeout, so I guess we can use `10s` here? 
Ideally, we should use the same timeout as user specified, but we can't get it from here.

We can start with `10s` and if it doesn't work well, we can reduce it further.

Reply:
Sounds good, it's just defensive coding in case the underlying d2 timeout is not triggered so we don't also end up exhausting the retry executor thread pool.